torch/_subclasses/functional_tensor.py

```diff
@@ -249,15 +248,32 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             )
             return NotImplemented
 
+        def _can_decompose(func):
+            # TODO (tmanlaibaatar)
+            # Eventually, we don't want to decompose any aten op at all
+            # except for aten ops that lie about their schemas (ops
+            # claimed to be functional but actually aren't).
+            # Since we never tested functionalization for every CompositeImplicitAutograd
+            # ops having a direct functional kernel. As a result, we can't safely
+            # turn off decomp for predispatch right now. Therefore, we do best
+            # effort by not decomposing ops that are functional in PreDispatch functionalization
+            # for now.
+            if self._dispatch_key is not None:
+                # only decompose view or inplace mutating ops
+                alias_info = len(
+                    [i for i in func._schema.arguments if i.alias_info is not None]
+                )
+                return alias_info != 0 or func._schema.is_mutable
```

Comment:
Agreed with the long term plan. For this PR though, I'm not sure that this is enough - you still still need to decompose "problematic" composite ops that functionalization can't handle. Otherwise, they will go through the functionalization fallback and do the wrong thing (the functionalization fallback can only handle "true" out-of-place/functional ops).

Reply:
Yeah makes sense sorry missed that part. Is it ok if I just have a list of two entries (batch_norm and native_batch_norm) for now?